### ISSUE TITLE
Cannot convert undefined or null to object on new code in sts.org (new SDK version 2.525)

### ISSUE THREAD

**Issue by andrewryan1906** [aws/aws-sdk-js#2839]
- [x] I've gone though [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x ] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js/issues) and didn't find any solution
  
**Describe the bug**
In my CI/CD pipeline when building my Node application using the new AWS SDK (version 2.525.0) I get this error:

```
Error: Cannot convert undefined or null to object

Stack: "TypeError: Cannot convert undefined or null to object\n    at hasOwnProperty (<anonymous>)\n    at features.constructor.validateRegionalEndpointsFlag (/rhythm/workspace/source/node_modules/aws-sdk/lib/services/sts.js:92:41)\n    at features.constructor.optInRegionalEndpoint (/rhythm/workspace/source/node_modules/aws-sdk/lib/services/sts.js:106:10)\n    at features.constructor.validateService (/rhythm/workspace/source/node_modules/aws-sdk/lib/services/sts.js:124:10)\n    at features.constructor.initialize (/rhythm/workspace/source/node_modules/aws-sdk/lib/service.js:52:10)\n    at features.constructor.Service [as constructor] (/rhythm/workspace/source/node_modules/aws-sdk/lib/service.js:40:10)\n    at features.constructor (/rhythm/workspace/source/node_modules/aws-sdk/lib/util.js:624:24)\n    at new features.constructor (/rhythm/workspace/source/node_modules/aws-sdk/lib/util.js:624:24)\n    at features.constructor.Service [as constructor] (/rhythm/workspace/source/node_modules/aws-sdk/lib/service.js:31:17)\n    at new features.constructor (/rhythm/workspace/source/node_modules/aws-sdk/lib/util.js:624:24)\n    at getCredentialsForRole (/rhythm/workspace/source/node_modules/@rhythmsoftware/aws/src/aws-helper.js:61:14)\n    at refreshCredentials (/rhythm/workspace/source/node_modules/@rhythmsoftware/aws/src/aws-helper.js:46:24)\n    at <anonymous>\n    at process._tickDomainCallback [as _tickCallback] (internal/process/next_tick.js:228:7)"
Exited with code 1
```

I can see the problem is with line 92 of a new chunk of code just added to the /lib/services/sts.js, where a null check is not being performed.

**Is the issue in the browser/Node.js?**
Node.js

**If on Node.js, are you running this on AWS Lambda?**
No.

**Details of the browser/Node.js version**

**SDK version number**
v2.525.0

**To Reproduce (observed behavior)**
A simple application run using this sdk seems to cause the error, though I haven't tried building an empty project.

**Expected behavior**
I expect to be able to run my Node application.

**Screenshots**
![image](https://user-images.githubusercontent.com/37671394/64559079-73abf780-d313-11e9-9973-3d187917ce58.png)

**Additional context**

**Comment by osubuu** [comment]
Also experiencing this issue

**Comment by AllanFly120** [comment]
Hi @andrewryan1906 @osubuu, the patch has been released in [2.526.0](https://github.com/aws/aws-sdk-js/blob/41fa25c023e90e137293adb50f312a25f4aa2d3d/lib/services/sts.js#L92). Please upgrade the SDK and have a try. Thank you for the bringing this up!

**Comment by jkachhia** [comment]
Thanks @AllanFly120
